### PR TITLE
Merge week 5 code into main

### PR DESCRIPTION
**Pull request by aidan-mundy**

### CODE REVIEW

File: src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -93,37 +81,50 @@ public RobotContainer() {
...
+    outtakeButton.whileHeld(new RunOuttakeCommand(outtake, -OuttakeConstants.SPEED));
 
-    towerButton.whileHeld(new RunTowerCommand(tower, TowerConstants.SPEED));
+    climberUpButton.whileHeld(new RunClimberCommand(climber, -ClimberConstants.SPEED));
+    climberDownButton.whileHeld(new RunClimberCommand(climber, ClimberConstants.SPEED));
 
-    climberUpButton.whileHeld(new RunClimberCommand(climber, ClimberConstants.SPEED));
-    climberDownButton.whileHeld(new RunClimberCommand(climber, -ClimberConstants.SPEED));
+    climberRetractButton.whileHeld(new RunClimberToPositionCommand(climber, 0));
+    climberExtendButton.whileHeld(new RunClimberToPositionCommand(climber, -170000));
```
Comment (citrus-melon) [inline src/main/java/frc/robot/RobotContainer.java]: Should these be extracted to a constant?

File: src/main/java/frc/robot/RobotContainer.java [hunk 2]
```diff
@@ -52,34 +50,24 @@
...
 
-  private final JoystickButton slowdownButton = driverJoystick.button(DriverButtons.SLOWDOWN);
+  private final JoystickButton slowdownButton1 = driverJoystick.button(DriverButtons.SLOWDOWN1);
+  private final JoystickButton slowdownButton2 = driverJoystick.button(DriverButtons.SLOWDOWN2);
 
-  private final JoystickButton intakeButton = operatorJoystick.button(OperatorButtons.INTAKE);
   private final JoystickButton outtakeButton = operatorJoystick.button(OperatorButtons.OUTTAKE);
-  private final JoystickButton towerButton = operatorJoystick.button(OperatorButtons.TOWER);
   private final JoystickButton climberUpButton = operatorJoystick.button(OperatorButtons.CLIMBER_UP);
   private final JoystickButton climberDownButton = operatorJoystick.button(OperatorButtons.CLIMBER_DOWN);
```
Comment (citrus-melon) [inline src/main/java/frc/robot/RobotContainer.java:68]: Maybe these should be called "spool" and "unspool"

File: src/main/java/frc/robot/Constants.java [hunk 3]
```diff
@@ -31,44 +31,50 @@
...
-      public static final int OUTTAKE = 2; // WEEK0
+      public static final int OUTTAKE = 1;
 
-      public static final int TOWER = 6; // WEEK0
+      public static final int TOWER = 2; // PLACEHOLDER
 
-      public static final int CLIMBER_UP = 4; // WEEK0
-      public static final int CLIMBER_DOWN = 1; // WEEK0
+      public static final int CLIMBER_UP = 6;
+      public static final int CLIMBER_DOWN = 5;
```
Comment (citrus-melon) [inline src/main/java/frc/robot/Constants.java:53]: Maybe these should be called "spool" and "unspool"

File: src/main/java/overclocked/stl/motorcontrol/OverclockedDifferentialDrive.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
+
+package overclocked.stl.motorcontrol;
+
+import edu.wpi.first.wpilibj.drive.DifferentialDrive;
+import edu.wpi.first.wpilibj.motorcontrol.MotorController;
+
+/** 
+ * An extension of the {@link DifferentialDrive} class that adds slew rate limiting.
+ * */
```
Comment (citrus-melon) [inline src/main/java/overclocked/stl/motorcontrol/OverclockedDifferentialDrive.java]:
```suggestion
 */
```

**Review by citrus-melon — DISMISSED** [review]

File: src/main/java/frc/robot/RobotContainer.java [hunk 5]
```diff
@@ -93,37 +81,50 @@ public RobotContainer() {
...
+    slowdownButton2.whileHeld(new TankDriveCommand(
         driveBase,
-        () -> DriveConstants.SLOWDOWN_PERCENT * driverJoystick.getRawAxis(DriverAxes.LEFT),
-        () -> DriveConstants.SLOWDOWN_PERCENT * driverJoystick.getRawAxis(DriverAxes.RIGHT)));
+        () -> DriveConstants.SLOWDOWN_PERCENT2 * driverJoystick.getRawAxis(DriverAxes.LEFT),
+        () -> DriveConstants.SLOWDOWN_PERCENT2 * driverJoystick.getRawAxis(DriverAxes.RIGHT)));
+
   }
 
   // A simple auto routine.
```
Comment (kchenturtles) [inline src/main/java/frc/robot/RobotContainer.java:113]: might need to fix the Java docs for auton

File: src/main/java/frc/robot/commands/climber/RunClimberToPositionCommand.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands.climber;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.Climber;
+
+public class RunClimberToPositionCommand extends CommandBase {
```
Comment (kchenturtles) [inline src/main/java/frc/robot/commands/climber/RunClimberToPositionCommand.java:13]: no javadocs

**Review by kchenturtles — CHANGES_REQUESTED** [review]
See comments -- just javadocs

File: src/main/java/frc/robot/RobotContainer.java [hunk 7]
```diff
@@ -148,22 +157,26 @@ public Command getAutonomousCommand() {
...
   public void setTeleopDefaultCommands() {
     driveBase.setDefaultCommand(
         new TankDriveCommand(
             driveBase,
             () -> driverJoystick.getRawAxis(DriverAxes.LEFT),
             () -> driverJoystick.getRawAxis(DriverAxes.RIGHT)));
+
+    // climber.setDefaultCommand(new ParallelCommandGroup(
+    // new RunOneClimberCommand(climber, () -> ClimberConstants.SPEED * operatorJoystick.getRawAxis(1), 0),
+    // new RunOneClimberCommand(climber, () -> ClimberConstants.SPEED * operatorJoystick.getRawAxis(5), 1)));
```
Comment (chit-sudeep) [inline src/main/java/frc/robot/RobotContainer.java]: delete as this code refers to a command that no longer exists

File: src/main/java/frc/robot/RobotContainer.java [hunk 8]
```diff
@@ -95,35 +83,54 @@ public RobotContainer() {
...
+          new TimedCommand(
+              AutonConstants.MEDIUM_AUTON_OUTTAKE_RUNTIME,
+              new RunOuttakeCommand(outtake, -OuttakeConstants.SPEED)),
+          new TimedCommand(
+              AutonConstants.SIMPLE_AUTON_RUNTIME,
+              new StraightDriveCommand(
+                  driveBase,
+                  AutonConstants.SIMPLE_AUTON_SPEED)));
+
+
```
Comment (chit-sudeep) [inline src/main/java/frc/robot/RobotContainer.java]:
```suggestion
// An auto routine that does nothing.
```

File: src/main/java/overclocked/stl/command/TimedCommand.java [hunk 9]
```diff
@@ -0,0 +1,22 @@
+
```
Comment (chit-sudeep) [inline src/main/java/overclocked/stl/command/TimedCommand.java:1]:
```suggestion
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.
```

File: src/main/java/overclocked/stl/motorcontrol/DampenedMotorController.java [hunk 10]
```diff
@@ -0,0 +1,117 @@
+
```
Comment (chit-sudeep) [inline src/main/java/overclocked/stl/motorcontrol/DampenedMotorController.java:1]:
```suggestion
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.
```

File: src/main/java/overclocked/stl/motorcontrol/OverclockedDifferentialDrive.java [hunk 11]
```diff
@@ -0,0 +1,52 @@
+
```
Comment (chit-sudeep) [inline src/main/java/overclocked/stl/motorcontrol/OverclockedDifferentialDrive.java:1]:
```suggestion
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.
```

File: src/main/java/overclocked/stl/command/TimedCommand.java (hunk 9 above)
Comment (aidan-mundy) [reply src/main/java/overclocked/stl/command/TimedCommand.java:1]: This is not copyright of FIRST/WPILib.

File: src/main/java/overclocked/stl/motorcontrol/DampenedMotorController.java (hunk 10 above)
Comment (aidan-mundy) [reply src/main/java/overclocked/stl/motorcontrol/DampenedMotorController.java:1]: This is not copyright of FIRST/WPILib.

File: src/main/java/overclocked/stl/motorcontrol/OverclockedDifferentialDrive.java (hunk 11 above)
Comment (aidan-mundy) [reply src/main/java/overclocked/stl/motorcontrol/OverclockedDifferentialDrive.java:1]: This is not copyright of FIRST/WPILib.

File: src/main/java/frc/robot/commands/drive/StraightDriveCommand.java [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+/**
+ * Drives a {@link DriveBase} in a (roughly) straight line.
+ */
+public class StraightDriveCommand extends TankDriveCommand {
+
+  /**
+   * Drives the driveBase in a (roughly) straight line.
+   *
+   * @param driveBase The {@link DriveBase} to drive
+   * @param speedSupplier Supplier forspeed
```
Comment (chit-sudeep) [inline src/main/java/frc/robot/commands/drive/StraightDriveCommand.java]:
```suggestion
   * @param speedSupplier Supplier for speed
```
I will approve the PR after this change is made.

**Review by chit-sudeep — APPROVED** [review]
I understand all of the changes.

**Review by kchenturtles — APPROVED** [review]
Changes make sense.

**Review by citrus-melon — APPROVED** [review]

**Pull request merged**